### DISCUSSION TITLE
r_picmip range

### DISCUSSION THREAD

**Discussion by Aciz** [etlegacy/etlegacy#1818] [Ideas]
Some people have asked if there's a reason why `r_picmip`is locked to 0-3 range. Well there really isn't, SD just changed the limit from 0-16 on RTCW to 0-3 for ET. Should we consider allowing higher values of picmip? Some comparison screenshots:

`r_picmip 3`, current max
![2021-11-09-161602-goldrush](https://user-images.githubusercontent.com/14221121/140940935-3929836e-29ea-4ba4-a9d3-7934a351f8c4.png)

`r_picmip 5`
![2021-11-09-161613-goldrush](https://user-images.githubusercontent.com/14221121/140941006-9afd6aef-4b4f-4015-83e9-7d3481990ba3.png)

`r_picmip 10`
![2021-11-09-161621-goldrush](https://user-images.githubusercontent.com/14221121/140941080-e543c598-4096-4cb1-85e4-9de9bcce3a9c.png)

`r_picmip 16` (doesn't seem to differentiate from 10 really)
![2021-11-09-161950-goldrush](https://user-images.githubusercontent.com/14221121/140941165-2a7ab266-7bfb-4ee1-b631-a70cc5240b54.png)

**Comment by jackeri** [comment]
The picmip values were clamped already in RTCW competitive due to them breaking the game with high picmip removing leafs and branches from trees and giving a really high visibility advantage to those using it.

**Reply by Aciz** [reply]
This has been fixed in ET by making the shaders used for leaves to have `noPicMip` as you can see in the images.

**Comment by jackeri** [comment]
On the other hand I also prefer forcing players to keep the game looking even something similar to what the map makers original design was since the main way to enjoy ET content in these days is twitch and other streaming services (outside of playing yourself of course). High picmips don't really do any favours to an already crappy looking game. I presume this somewhat the same reason why most other games these days with competitive scenes limit the graphical settings. I guess this is something that the whole team should give some feedback on.

**Reply by x0rnn** [reply]
> I guess this is something that the whole team should give some feedback on.

@Aranud @rafal1137 your opinions? Also @realkemon your opinion as a map maker?

**Comment by ETLIssues** [comment]
Since there are no problems with foliage disappearing, this seems like a simple decision to unlock. The game should not be designed based on what looks good on a stream.

**Comment by oheikk** [comment]
On the one hand, configs shouldn't provide a crazy visibility advantage as per conventions set by many modern games (Jacker makes a very good argument about this sort of thing).

On the other hand, ET/ETL already offers many options that improve visibility and general information gain if a player is good at using their config, and r_picmip 3 just feels like an extremely arbitrary limit.

IMO consider how this affects mods other than Legacy. If it does, don't change anything. If it can be done without affecting other mods, do it. :)

**Reply by ETLIssues** [reply]
> On the one hand, configs shouldn't provide a crazy visibility advantage as per conventions set by many modern games (Jacker makes a very good argument about this sort of thing).

This is a fair argument, but I think it's addressed completely by the noPicMip implementation that Aciz describes. Or at least, I can't think of any scenarios where you'd be helped with the lower graphic quality. Definitely open to counter-examples.

EDIT: server admins and league configs could still easily force whatever settings are deemed reasonable by admins

**Comment by jackeri** [comment]
You can already downgrade the graphics to look like a dos game, and imho enough is enough. It's 2023 not 1993.

**Comment by Oksii** [comment]
> EDIT: server admins and league configs could still easily force whatever settings are deemed reasonable by admins

This. Why take authority over something that was always decided by the respective leagues and tourneys in the past? 

> You can already downgrade the graphics to look like a dos game, and imho enough is enough. It's 2023 not 1993.

This is such a ridiculous argument to make and hill to die on. Games made in 2023 and 2024 often chase the simplistic and minimal art style, pixel art or otherwise that mimic dos games. 

> since the main way to enjoy ET content in these days is twitch and other streaming services (outside of playing yourself of course).

Just because you subjectively don't find something appealing doesn't mean others don't. 
I'd much prefer to have clean visuals with minimal textures on my twitch streams, even just for the sake of dealing with twitch's ridiculously low bitrate limit it would produce far better looking streams for most people. 

It's a 20 year old game, absolutely no one is playing it for its visuals or graphic fidelity, no one in gathers or comp are playing maps for their decorative backgrounds, neither is anyone that's viewing these streams paying attention to it. They're here for the gameplay. 

Give the authority back to the server admins where it belongs, they and their demographic can choose what they prefer.

**Reply by jackeri** [reply]
As far as I am aware, the server admins have all the power in the world on these things. They can force cvar values and they can even use custom pk3 files to just plain replace all textures with a single pixel color replacements. Effectively having a picmip 100.

But to actually respond to your point, higher picmic gives you a lot more visibility, which does mean to have a even playing field it would just be better to enforce a higher picmip range for everyone.

The competitive scene is and has always been different in a sense that they had (for the most part) their configs dialed in from day 1, where as the public scene still has many or even most players running somewhat default configs graphics vise. The comp scene is extremely important to the game, no doubth about that, but it is not the only people playing the game.

**Reply by Oksii** [reply]
> As far as I am aware, the server admins have all the power in the world on these things. They can force cvar values

r_picmip just like com_maxfps, and presumably other cvars, are hardcoded in the client. No server configs allow us to bypass or overwrite these limits. Hence this discussion even existing.

**Reply by jackeri** [reply]
Ah, right you were, I mixed the cheat protected cvars with the locked one's. Does not change my point, though.

**Comment by iamez** [comment]
beautifull screenshots, feels like the game is finally improved and something usefull was added (something someone(s) are asking about for some time)

would be nice to have this implemented
